### PR TITLE
DD-262 Line references for code blocks

### PR DESCRIPTION
**Pull request by ccgeller**
Jira: [DD-262](https://circleci.atlassian.net/jira/software/c/projects/DD/boards/477?modal=detail&selectedIssue=DD-262)
Preview link: [link](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/DD-262-add-line-references-for-code-blocks-preview/?force-all)

# Description
Have the code lines numbered so it’s easier for the docs team and end-user to refer to specific lines of a code block. 

# Reasons
Improve the end-user / docs teams' experience when referencing and writing the code

# Changes
- Add new library[ Line numbering plugin for Highlight.js by Yauheni Pakala ](https://wcoder.github.io/highlightjs-line-numbers.js/)
- Remove linenos (old way of adding in line numbers)

# Todo
- [x] more width options for long numbers ect. 100, 1000
Screen Shot 2021-12-13 at 8 21 39 PM

# Before
Screen Shot 2021-12-13 at 11 25 36 AM

# After
Screen Shot 2021-12-13 at 11 26 08 AM

# Before linenos
Screen Shot 2021-12-13 at 11 27 28 AM

# After linenos removed
Screen Shot 2021-12-13 at 11 27 41 AM

# Before single line
Screen Shot 2021-12-13 at 8 25 59 PM

# After single line
Screen Shot 2021-12-13 at 8 26 19 PM

Adding number for single line been reviewed with @danbeere

### CODE REVIEW

**Review by teesloane — CHANGES_REQUESTED** [review]
Nice work! Everything looks good in the preview link.

One request: it looks like you have installed the package locally in package.json and the yarn.lock file has been updated -- but the package is also being included in `jekyll/_includes/head.html`. I think we should probably be only doing one of these things (the package.json version, I think).

File: src/styles/main.scss [hunk 1]
```diff
@@ -299,6 +299,27 @@ div.align-center img{
...
 }
 
+/* highlightjs number block of numbers */
+table tr td.hljs-ln-numbers {
+    -webkit-touch-callout: none;
+    -webkit-user-select: none;
+    -khtml-user-select: none;
+    -moz-user-select: none;
+    -ms-user-select: none;
+    user-select: none;
```
Comment (tshen815) [inline src/styles/main.scss:309]: is this to prevent users from being able to select the numbers in all different browsers?

Comment (ccgeller) [reply src/styles/main.scss:309]: Yes! https://developer.mozilla.org/en-US/docs/Web/CSS/user-select

**Review by teesloane — APPROVED** [review]
Looks good!

**Review by rsylvian — APPROVED** [review]
Great job 👍

**Review by devinmarieb — APPROVED** [review]
Nice thanks!

**Review by heug — APPROVED** [review]

**Pull request merged**